CHANGELOG.md

```diff
@@ -0,0 +1,40 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
+and follows a [Backwards Compatability Policy]
+(https://docs.solana.com/developing/backwards-compatibility)
+
+## Adding to this Changelog
+### Audience
+* Entries in this log are intended to be easily understood by consensus
+validators, rpc operators, and dapp developers.
+
+### Noteworthy
+* Changes that relate to a SIMD, or add a feature gate should add an entry to
+this log. Other changes that are relevant to the aforementioned audiences should
+add an entry at the discretion of authors and reviewers.
+
+### Instructions
+* Update this log in the same pull request that implements the change. If the
+change is spread over several pull requests update this log in the one that
+makes the feature code complete.
+* Add notes to the [Unreleased] section in each branch that you merge to.
```

Comment:
Is the `Unreleased` section features that would not get immediately released in each version?

Reply:
So this would be something like RuntimeV2 which has a ton a subtasks but would live in `Unreleased` until all subtasks are complete?

Reply:
https://keepachangelog.com/en/1.0.0/  envisions [Unreleased] to just be the notes for the next version. If we adopt that practice then I like @mvines suggestion of just numbering it when we bump the version number.

I don't think we need a section for partially complete work, since we already have guidance to add a log entry when the work is code complete.